background.ts

```diff
@@ -218,6 +218,9 @@ Electron.app.whenReady().then(async() => {
       const message = err.message || err.toString();
 
       showErrorDialog(titlePart, message, true);
+      if (err instanceof settings.SettingsError) {
```

Comment:
We're passing in `fatal=true`, and the app quits immediately; please add a comment on why the extra `return` is required (because it's very non-obvious).

Reply:
We've been using `Electron.app.quit()` which shuts things down gracefully, but the code is still running asynchronously. Maybe we should create an issue to use `Electron.app.exit` instead?

Reply:
Okay, in that case: why do we _not_ need to do the early return if `err` is _not_ `SettingsError` (where we still pass `fatal=true`)?

That is: I don't understand why the condition exists (rather than just always returning early).

Reply:
Because the current JavaScript code block is still executing. Once it's done, the shutdown process will start. This is what the output looks like without the return statement:
```
Error

No version specified in /Users/ericp/Library/Preferences/rancher-desktop/settings.json
2023-11-28T21:42:35.666Z: Error starting up: TypeError: Cannot read properties of undefined (reading 'application') TypeError: Cannot read properties of undefined (reading 'application')
    at /Users/ericp/git/ericpromislow/rancher/desktop/dist/app/background.js:44122:113
```

The `return` stmt prevents the `Cannot read properties of undefined` lines from appearing.  These message are appearing because `settings` is unset.

The app is going to shut down anyway, and those lines just confuse things.

Reply:
Okay, why don't we need it for all the _other_ cases? What makes this one special? (And that needs to live in the code, not in a PR comment…)

Reply:
We really needed to return in this case, because `cfg` is null. I explain that, but I added a `return` for the other class of fatal errors because there's no point continuing to run the `whenReady` handler for them as well.